File: Search/Linear Search/src.py
"""
ALGORITHM : Binary Search
WORST CASE => {
    PERFORMANCE:  O(n)
    SPACE:        O(1)
}
"""


def linear_search(arr, key):
    size = len(arr)
    for i in range(size):
        if arr[i] == key:
            return True, i

    return False, None


if __name__ == '__main__':
    a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
    found, index = linear_search(a, 4)

    if found:
        print("Found at position {}".format(index))
    else:
        print("Not Found!")

File: Algorithms/Search/Linear Search/src.java
/*
ALGORITHM : Linear Search
WORST CASE => {
    PERFORMANCE:  O(n)
    SPACE:        O(1)
}
*/

import java.util.Scanner;

public class LinearSearch {
    public static void  main(String[] args) {
        Scanner myObj = new Scanner(System.in);

        int[] arr;
        arr = new int[] {3, 8, 1, 7, 5, 9, 2};

        System.out.print("Enter Element to find: ");
        int key = myObj.nextInt();
        for (int element : arr) {
            if (element == key) {
                System.out.println("Found!");
                return;
            }
        }
        System.out.println("Not Found!");
    }
}
